ODIN_II/SRC/YYosys.cpp

```diff
@@ -259,14 +259,14 @@ void YYosys::execute() {
             run_pass(std::string("hierarchy -check -auto-top -purge_lib"));
         }
 
-        // Use a readable name convention
-        run_pass(std::string("autoname"));
         // Translate processes to netlist components such as MUXs, FFs and latches
         run_pass(std::string("proc; opt;"));
         // Extraction and optimization of finite state machines
         run_pass(std::string("fsm; opt;"));
         // Collects memories, their port and create multiport memory cells
         run_pass(std::string("memory_collect; memory_dff; opt;"));
+        // Use a readable name convention
```

Comment:
It seems like it would be good to comment that this step can have a high memory footprint (can reference the issue #), and hence it is best to run it after the optimizations above as the smaller netlist helps.  
 --> Hopefully my understanding is correct, if not please correct in the comment :)

Reply:
I will certainly add related comments, and yes, your statement is entirely correct.